File: 🟢 Easy/94 - Binary Tree Inorder Traversal/_94BinaryTreeInorderTraversal.java
class Solution {
    public static void inOrder(TreeNode root, ArrayList<Integer> arr) {
        if (root == null) {
            return;
        }
        inOrder(root.left, arr);
        arr.add(root.val);
        inOrder(root.right, arr);
    }

    public List<Integer> inorderTraversal(TreeNode root) {
        ArrayList<Integer> al = new ArrayList<>();
        inOrder(root, al);
        return al;
    }
}

File: 🟢 Easy/94 - Binary Tree Inorder Traversal/_94BinaryTreeInorderTraversal.cpp
class Solution {
public:
    void inOrder(TreeNode* root, vector<int>& res) {
        if (root == NULL) {
            return;
        }
        inOrder(root->left, res);
        res.push_back(root->val);
        inOrder(root->right, res);
    }

    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> res;
        inOrder(root, res);
        return res;
    }
};